plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+		},
+		{
+			name: "valid wildcard_host",
+			config: `{
+				"wildcard_host": "*test.com",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid wildcard_host",
+			config: `{
+				"wildcard_host": "test.*",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid wildcard_host with multiple *",
+			config: `{
+				"wildcard_host": "*test.*",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"wildcard_host":"only one wildcard must be specified"}}`,
+		},
+		{
+			name: "invalid wildcard_host with * in the middle",
+			config: `{
+				"wildcard_host": "new*test",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"wildcard_host":"wildcard must be leftmost or rightmost character"}}`,
+		},
+		{
+			name: "valid ipv4",
+			config: `{
+				"ip": "192.0.2.1",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ipv4",
+			config: `{
+				"ip": "192.0.2.1.1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 192.0.2.1.1"}}`,
+		},
+		{
+			name: "valid ipv6",
+			config: `{
+				"ip": "2001:db8::1",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ipv6",
+			config: `{
+				"ip": "2001:db8::1::1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 2001:db8::1::1"}}`,
+		},
+		{
+			name: "valid ip_or_cidr",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ip_or_cidr with too big mask",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip_or_cidr":"invalid ip or cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid cidr_v4",
+			config: `{
+				"cidr_v4": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid cidr_v4 with too big mask",
+			config: `{
+				"cidr_v4": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"cidr_v4":"invalid ipv4 cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid port",
+			config: `{
+				"port": 80,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid port as string",
+			config: `{
+				"port": "80",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"expected an integer"}}`,
+		},
+		{
+			name: "invalid too big port",
+			config: `{
+				"port": 9999999999,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
+		{
+			name: "invalid negative port",
+			config: `{
+				"port": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
+		{
+			name: "valid url",
+			config: `{
+				"url": "https://www.konghq.com",
+				"tag": "test"
+			}`,
```

Comment:
We should ad a negative test here; e.g. something with a space (unless things are escaped I forget) and maybe a relative path.

Reply:
It seems that spaces are somehow accepted today :\

```
$ echo '
name: http-log
config:
    http_endpoint: http://www.konghq .com
' | y2j | http put ":8001/plugins/9c8047db-b57e-43a9-8a41-dd83a302a1b8"

{
    "config": {
        "content_type": "application/json",
        "custom_fields_by_lua": null,
        "flush_timeout": 2,
        "headers": null,
        "http_endpoint": "http://www.konghq .com",
        "keepalive": 60000,
        "method": "POST",
        "queue_size": 1,
        "retry_count": 10,
        "timeout": 10000
    },
    "consumer": null,
    "created_at": 1652175943,
    "enabled": true,
    "id": "9c8047db-b57e-43a9-8a41-dd83a302a1b8",
    "name": "http-log",
    "protocols": [
        "grpc",
        "grpcs",
        "http",
        "https"
    ],
    "route": null,
    "service": null,
    "tags": null
}
```

Reply:
https://github.com/Kong/goks/pull/14/commits/86049a28f32931188a6f4e698d3acf15aa30df94

Reply:
> It seems that spaces are somehow accepted today

Make sense, they are probably going through the escaping function.